app/controllers/samples_controller.rb

```diff
@@ -222,6 +231,57 @@ def change_ownership(new_owner)
     end
   end
 
+  def create_qc_info(sample)
```

Comment:
These 3 methods (`create_qc_info`, `create_notes_for_qc_info` and `create_assay_attachments_for_qc_info`) have little to do with the application logic, which only cares about: duplicate this data unless it has already. Since they only manipulate data, they would fit better into the model. As a bonus the names would be more explicit or more succinct. For example:

```ruby
class QcInfo
  def self.find_or_duplicate_from(sample)
  end

  def self.duplicate_notes(qc_info, sample)
  end

  def self.duplicate_assay_attachments(qc_info, sample)
  end
end
```

Then in the controller we can merely:

```ruby
to_transfer.qc_info = QcInfo.find_or_create_from(to_transfer)
to_transfer.old_batch_number = to_transfer.batch.batch_number
```

Nothing more.